core/common/src/main/scala/org/typelevel/otel4s/TextMapPropagator.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+import cats.Applicative
+import org.typelevel.vault.Vault
+
+trait TextMapPropagator[F[_]] {
```

Comment:
```suggestion
@annotation.implicitNotFound("""
Could not find the `TextMapPropagator` for ${F}. The `TextMapPropagator` can be materialized:

1) Manually from the `Otel4s` instance

val otel4s: org.typelevel.otel4s.Otel4s[${F}] = ???
implicit val propagator: TextMapPropagator[${F}] = otel4s.propagators.textMapPropagator

2) Implicitly from the implicit `Otel4s` instance

implicit val otel4s: org.typelevel.otel4s.Otel4s[${F}] = ???
implicit val propagator: TextMapPropagator[${F}] = implicitly
""")
trait TextMapPropagator[F[_]] {
```

Reply:
That's a real good idea.  `TextMapGetter` and `TextMapSetter` would benefit from these for defining "carrier types," too.

Reply:
This is still a good suggestion, but we don't have that implicit yet.  I'm making note of this so we don't lose it.